DSEGOG-9 user can remove columns with close icon and middle mouse click

sam-glendenning opened the pull request:
## Description
The user is now able to more easily remove columns from display. A black cross icon is displayed in each column header which the user can click to remove the column. I also added in the ability for the user to click the entire column header with the middle mouse button as an alternative method for removing the column from display. This mimics the behaviour of popular web browsers, where a user can middle click a tab to close it. This may be a familiar action for some users and speed up their interaction with the site.

## Testing instructions
Try removing columns using both methods and test that the functionality works as expected. Try this across multiple pages to double check it's working. Also make sure that, when a column is closed, its checkbox is unticked at the bottom of the page.

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes [DSEGOG-9](https://stfc.atlassian.net/jira/software/projects/DSEGOG/boards/22?selectedIssue=DSEGOG-9)

Commit:
DSEGOG-9 adding icon to remove column on click

This appears underneath the column name at the moment but will be moved

Commit:
DSEGOG-9 Styling close icon and adding middle click to close functionality

The close icon is now styled so that, on hover, it changes from black to red in color and vice versa

Also adding functionality so that the middle mouse button can be used to remove the column (can use middle click across the entire data header). This mimics the functionality of closing a tab in a web browser, which may be a familiar and convenient action for some users

Commit:
Merge branch 'develop' into DSEGOG-9-user-can-remove-columns

Commit:
DSEGOG-9 adding unit tests for column closing

Includes tests for dataHeader and recordTable components

Commit:
DSEGOG-9 improving unit test functionality

Improving on methods for fetching elements on the screen by adding aria-labels and data-testids

louise-davies reviewed (changes requested):
Code looks good - I don't like that there's an extra floating line under the headers (see screenshot) - can you try and remove that? I think the issue is that there's two `TableCell`s

![image](https://user-images.githubusercontent.com/22525471/173087815-3bd757b4-606e-44d9-8b42-ef1a43b5b7fc.png)

Commit:
DSEGOG-9 removing extra horizontal line in column header

This occurred due to nested <TableCell> elements. Updated the DataHeader to be a <TableCell> th component to ensure proper DOM nesting in snapshot tests

sam-glendenning commented on src/table/headerRenderers/dataHeader.component.test.tsx:
```diff
@@ -13,7 +13,15 @@ describe('Data Header', () => {
 
   const createView = (): RenderResult => {
-    return render(<DataHeader {...props} />);
+    return render(
+      <table>
+        <thead>
+          <tr>
+            <DataHeader {...props} />
+          </tr>
+        </thead>
+      </table>
+    );
```
This tests the column header in a proper tabular context while also removing the error of invalid DOM nesting in snapshot tests
